Source/Spring2022_Capstone/Weapon/WeaponBase.cpp

```diff
@@ -96,6 +99,14 @@ void AWeaponBase::WeaponCooldown()
 	bIsOverheating = false;
 	bCanFire = true;
 	CurWeaponCharge = 0;
+	if (OnWeaponOverheatChangedDelegate.IsBound())
+	{
+		OnWeaponOverheatChangedDelegate.Execute(false);
+	}
+	if (OnWeaponChargeChangedDelegate.IsBound())
```

Comment:
There's no need to check if it's bound. Not sure if it's the same Execute() call but check on documentation to fix this.
Yes, I made the same mistake before and have kept an eye on it to fix it.